src/data_structures/two_pointer.md

```diff
@@ -0,0 +1,80 @@
+# Two Pointer Algorithm
+
+## Overview
+
+The Two Pointer Algorithm is a technique used for efficiently solving problems with linear data structures like arrays or linked lists. It involves using two pointers to traverse the data structure and perform operations based on certain conditions.
+
+## Basic Idea
+
+The basic idea behind the Two Pointer Algorithm is to maintain two pointers that traverse the data structure in a way that reduces the time complexity. The pointers move towards each other or in a specific pattern based on the problem requirements.
+
+## Algorithm Steps
+
+1. **Initialize Pointers:**
+   - Initialize two pointers, usually at the beginning or end of the data structure.
+
+2. **Move Pointers:**
+   - Move the pointers towards each other, closer or farther based on the problem requirements.
+   - Update pointers based on specific conditions.
+
+3. **Check Conditions:**
+   - Check conditions at each step to determine if the solution is found or if the pointers need to be adjusted.
+
+4. **Repeat:**
+   - Repeat steps 2 and 3 until the pointers meet or fulfill the problem requirements.
+
+## Common Use Cases
```

Comment:
Please give specific details on how you use 2-pointers to solve each of the followings. You should assume people who are reading this doesn't know how 2-pointers works beforehand.